Source/Xamarin/Prism.DryIoc.Forms/DryIocExtensions.cs

```diff
@@ -11,34 +12,150 @@ namespace Prism.DryIoc
+        /// <param name="DesktopView">Desktop Specific View Type</param>
+        /// <param name="TabletView">Tablet Specific View Type</param>
+        /// <param name="PhoneView">Phone Specific View Type</param>
+        public static void RegisterTypeForNavigationOnIdiom<TView, TViewModel>(this IContainer container, string name = null, Type DesktopView = null, Type TabletView = null, Type PhoneView = null)
+            where TView : Page
+            where TViewModel : class
+        {
+            if (Device.Idiom == TargetIdiom.Desktop && DesktopView != null)
+            {
+                container.RegisterTypeForNavigationWithViewModel<TViewModel>(DesktopView, name);
+            }
+            else if (Device.Idiom == TargetIdiom.Phone && PhoneView != null)
+            {
+                container.RegisterTypeForNavigationWithViewModel<TViewModel>(PhoneView, name);
+            }
+            else if (Device.Idiom == TargetIdiom.Tablet && TabletView != null)
+            {
+                container.RegisterTypeForNavigationWithViewModel<TViewModel>(TabletView, name);
+            }
+            else
+            {
+                container.RegisterTypeForNavigation<TView, TViewModel>(name);
+            }
+        }
+
+        //TODO: decide if Prism will continue to support navigating via ViewModels, or should we stick to just a single approach to navigation
+        //public static void RegisterTypeForViewModelNavigation<TView, TViewModel>(this IContainer container)
+        //    where TView : Page
+        //    where TViewModel : class
+        //{
+        //    container.RegisterTypeForNavigationWithViewModel<TViewModel>(typeof(TView), typeof(TViewModel).FullName);
+        //}
+
+        //public static void RegisterTypeForViewModelNavigationOnPlatform<TView, TViewModel>(this IContainer container, Type AndroidView = null, Type iOSView = null, Type OtherView = null, Type WindowsView = null, Type WinPhoneView = null)
+        //    where TView : Page
+        //    where TViewModel : class
+        //{
+        //    container.RegisterTypeForNavigationOnPlatform<TView, TViewModel>(typeof(TViewModel).FullName, AndroidView, iOSView, OtherView, WindowsView, WinPhoneView);
+        //}
+
+        //public static void RegisterTypeForViewModelNavigationOnIdiom<TView, TViewModel>(this IContainer container, Type DesktopView = null, Type TabletView = null, Type PhoneView = null)
+        //    where TView : Page
+        //    where TViewModel : class
+        //{
+        //    container.RegisterTypeForNavigationOnIdiom<TView, TViewModel>(typeof(TViewModel).FullName, DesktopView, TabletView, PhoneView);
+        //}
+
+        private static void RegisterTypeForNavigationWithViewModel<TViewModel>(this IContainer container, Type viewType, string name = null)
+            where TViewModel : class
```

Comment:
Shouldn't this be BindableBase? Vaguely recall that @dansiegel changed from class to BindableBase in Unity